crates/utils/src/utils/validation.rs

```diff
@@ -158,31 +158,27 @@ pub fn is_valid_post_title(title: &str) -> LemmyResult<()> {
 }
 
 /// This could be post bodies, comments, or any description field
-pub fn is_valid_body_field(body: &Option<String>, post: bool) -> LemmyResult<()> {
-  if let Some(body) = body {
-    if post {
-      max_length_check(body, POST_BODY_MAX_LENGTH, LemmyErrorType::InvalidBodyField)?;
-    } else {
-      max_length_check(body, BODY_MAX_LENGTH, LemmyErrorType::InvalidBodyField)?;
-    };
-  }
+pub fn is_valid_body_field(body: &str, post: bool) -> LemmyResult<()> {
```

Comment:
We already have a lot of `if let Some(...` in the API code, so I've removed a few of these internal if let wrappers, as they're pretty pointless. Makes sense to just work with the data directly in a lot of cases.